Assets/Scripts/Source/Game/Combat/CombatConfig.cs

```diff
@@ -1,7 +1,27 @@
 using UnityEngine;
+using System;
 
-public abstract class AbstractCombatConfig : MonoBehaviour
+public class CombatConfig : AbstractCombatConfig
 {
-    public abstract Player GetPlayer();
-    public abstract Enemy[] GetEnemies();
+    Player player;
+    Enemy enemy;
+
+    public void Initialize(Player savedPlayer, Enemy nextEnemy)
+    {
+        player = savedPlayer;
```

Comment:
Could we build this logic into `Character::CopyConfig` and `Enemy::CopyConfig` instead?